sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/ExponentialCounterTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.state;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class ExponentialCounterTest {
+
+  static Stream<ExponentialCounterFactory> counterProviders() {
+    return Stream.of(
+        ExponentialCounterFactory.mapCounter(), ExponentialCounterFactory.circularBufferCounter());
+  }
+
+  @ParameterizedTest
```

Comment:
There is one or two more tests we need once we have sufficient implementations of `BucketMapper` - since the current implementation can be inaccurate in certain cases. I can get to that in the next couple of weeks.

Reply:
That'd be great.  I was looking at the Go code (which I based this on), and there's some big complex optimisations to half the cost of the log lookup.   That may be the next thing in the critical path here.

Reply:
This is the ticket: https://github.com/open-telemetry/opentelemetry-java/issues/3842